catalyst-toolbox/src/rewards/proposers/util.rs

```diff
@@ -0,0 +1,70 @@
+use std::{
+    borrow::Cow,
+    path::{Path, PathBuf},
+};
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+pub fn build_path_for_challenge(path: &Path, challenge_name: &str) -> PathBuf {
+    let challenge_name = sanitize_name(challenge_name);
+    let ext = path.extension();
+
+    let mut path = path.with_extension("").as_os_str().to_owned();
+    path.push("_");
+    path.push(&*challenge_name);
+    let path = PathBuf::from(path);
+
+    match ext {
+        Some(ext) => path.with_extension(ext),
+        None => path,
+    }
+}
+
+fn sanitize_name(name: &str) -> Cow<'_, str> {
```

Comment:
https://github.com/input-output-hk/catalyst-toolbox/pull/146/files#diff-0b91cc9037e52e3f47667ccf7a7f8f61a6f27a42f33a854c6adb3de87568706aR24 seems like replication? is the other PR still relevant?

Reply:
Huh, not quite sure what's happened with these PRs. This one must have got mixed up with the rust rewrite at some point :shrug: